src/mlpack/methods/preprocess/image_converter_main.cpp

```diff
@@ -0,0 +1,132 @@
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core.hpp>
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+using namespace mlpack::data;
+
+
+PROGRAM_INFO("Image Converter",
+    // Short description.
+    "A utility to load an image or set of images into a single dataset that"
+    "can then be used by other mlpack methods and utilities. This can also"
+    "unpack an image dataset into individual files.",
+    // Long description.
+    "This utility takes a image or an array of images and loads them to a"
+    " matrix. You can specify the height " + PRINT_PARAM_STRING("height") +
+    " width " + PRINT_PARAM_STRING("width") + " and channel " +
+    PRINT_PARAM_STRING("channel") + " of the images that needs to be loaded. "
+    "\n"
+    "There are other options too, that can be specified such as " +
+    PRINT_PARAM_STRING("quality")
+    + ".\n\n" +
+    "You can also provide a dataset and save them as images using " +
+    PRINT_PARAM_STRING("dataset") + " and " + PRINT_PARAM_STRING("save") +
+    " as an parameter. An example to load an image : "  +
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "output", "Y") +
+    "\n\n" +
+    " An example to save an image is :" +
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "dataset", "Y", "save", true),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// DEFINE PARAM
+PARAM_VECTOR_IN_REQ(string, "input", "Image filenames which have to "
+    "be loaded/saved.", "i");
+
+PARAM_INT_IN("width", "Width of the image", "W", 256);
+PARAM_INT_IN("channel", "Number of channel", "C",  3);
+
+PARAM_MATRIX_OUT("output", "Matrix to save images data to.", "o");
+
+PARAM_INT_IN("quality", "Compression of the image if saved as jpg (0-100).",
+    "q", 90);
+
+
+PARAM_INT_IN("height", "Height of the images", "H", 256);
+PARAM_FLAG("save", "Save a dataset as images", "s");
+PARAM_MATRIX_IN("dataset", "Input matrix to save as images.", "I");
+
+// Loading/saving of a Image Info model.
+PARAM_MODEL_IN(ImageInfo, "input_model", "Input Image Info model.", "m");
+PARAM_MODEL_OUT(ImageInfo, "output_model", "Output Image Info model.", "M");
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  data::ImageInfo* info;
+
+  Timer::Start("Loading/Saving Image");
+  if (CLI::HasParam("input_model"))
+  {
+    info = CLI::GetParam<ImageInfo*>("input_model");
+  }
+  else
+  {
+    if (!CLI::HasParam("width") || !CLI::HasParam("height") ||
+        !CLI::HasParam("channel"))
+    {
+      throw std::runtime_error("Please provide height, width and "
+          "number of channels of the images.");
+    }
```

Comment:
I think that we should adapt this code to automatically detect these parameters from the images if they aren't specified, so also the checks below won't be necessary.

Reply:
How should I automatically detect it ?